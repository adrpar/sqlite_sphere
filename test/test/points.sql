.load ../build/libsqlitesphere.so

-- operators
SELECT sequal( spoint( '(0, 90d)' )  , spoint( '(0, 90d)' ) );
SELECT sequal( spoint( '(0, 90d)' )  , spoint( '(0,-90d)' ) );
SELECT sequal( spoint( '(0,-90d)' )  , spoint( '(0,-90d)' ) );
SELECT snotequal( spoint( '(0, 90d)' ) , spoint( '(0, 90d)' ) );
SELECT snotequal( spoint( '(0, 90d)' ) , spoint( '(0,-90d)' ) );
SELECT snotequal( spoint( '(0,-90d)' ) , spoint( '(0,-90d)' ) );
SELECT sequal( spoint( '(0d,  0)' )  , spoint( '(360d,0)' ) );

-- I/O test --

-- Output ---

SELECT sstr( spoint( '( 1h 2m 30s , +1d 2m 30s)' ), "DEG", 8 );

SELECT sstr( spoint( '( 0h 2m 30s , +0d 2m 30s)' ), "DEG", 8 );

SELECT sstr( spoint( '( 1h 3m 30s , -1d 3m 3.6s)' ), "DEG", 8 );

SELECT sstr( spoint( '( 0h 3m 30s , -0d 3m 3.6s)' ), "DEG", 8 );

SELECT sstr( spoint( '( 0h 2m 30s , 90d 0m 0s)' ), "DEG", 8 );

SELECT sstr( spoint( '( 0h 2m 30s ,-90d 0m 0s)' ), "DEG", 8 );


SELECT sstr( spoint( '( 1h 2m 30s , +1d 2m 30s)' ), "DMS", 8 );

SELECT sstr( spoint( '( 0h 2m 30s , +0d 2m 30s)' ), "DMS", 8 );

SELECT sstr( spoint( '( 1h 2m 30s , -1d 2m 30s)' ), "DMS", 8 );

SELECT sstr( spoint( '( 0h 2m 30s , -0d 2m 30s)' ), "DMS", 8 );

SELECT sstr( spoint( '( 0h 2m 30s , 90d 0m 0s)' ), "DMS", 8 );

SELECT sstr( spoint( '( 0h 2m 30s ,-90d 0m 0s)' ), "DMS", 8 );



SELECT sstr( spoint( '( 1h 2m 30s , +1d 2m 30s)' ), "HMS", 8 );

SELECT sstr( spoint( '( 0h 2m 30s , +0d 2m 30s)' ), "HMS", 8 );

SELECT sstr( spoint( '( 1h 2m 30s , -1d 2m 30s)' ), "HMS", 8 );

SELECT sstr( spoint( '( 0h 2m 30s , -0d 2m 30s)' ), "HMS", 8 );

SELECT sstr( spoint( '( 0h 2m 30s , 90d 0m 0s)' ), "HMS", 8 );

SELECT sstr( spoint( '( 0h 2m 30s ,-90d 0m 0s)' ), "HMS", 8 );



SELECT sstr( spoint( '( 0h 0m 0s , 0d 0m 0s)' ), "RAD", 8 );

-- "incorrect dec. values"



SELECT sstr( spoint( '( 0h 2m 30s , 95d 0m 0s)' ), "DEG", 8 );

SELECT sstr( spoint( '( 24h 2m 30s , 5d 0m 0s)' ), "DEG", 8 );

SELECT sstr( spoint( '( -0h 2m 30s , -5d 0m 0s)' ), "DEG", 8 );

SELECT sstr( spoint( '( 0h 2m 30s , -95d 0m 0s)' ), "DEG", 8 );

-- Input --

SELECT sstr( spoint( '( 0.625d , -0.04166666667d)' ), "DEG", 8 );

SELECT sstr( spoint( '(0.0109083078249646 , -0.000727220521664407)' ), "DEG", 8 );

-- functions for point --------------

-- sstr( spoint(float8, float8 ))

SELECT sstr( spoint(0.0109083078249646 , -0.000727220521664407), "DEG", 8 );



SELECT sstr( spoint(7.28318530717958623 , 0.00), "RAD", 8 );

SELECT sstr( spoint(0.0 , 2.141592653589793116), "RAD", 8 );

-- sdist(spoint, spoint)

SELECT sdist(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 0m 30s , 0d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) );

SELECT sdist(spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

-- spoint_long(spoint)

SELECT spoint_long(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,95d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,85d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,-95d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,-85d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,90d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s , 0d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,95d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,85d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,-95d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,-85d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,90d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,-90d 0m 0s)' ) );

-- spoint_lat(spoint)

SELECT spoint_lat(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,95d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,85d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,-95d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,-85d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,90d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s , 0d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,95d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,85d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,-95d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,-85d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,90d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,-90d 0m 0s)' ) );

-- xyz(spoint)

SELECT spoint_x(spoint( '(0d,-90d)' ) );

SELECT spoint_y(spoint( '(0d,-90d)' ) );

SELECT spoint_z(spoint( '(0d,-90d)' ) );

-- operators for points -------------

-- opsequal( erator ---------------------- )-

SELECT sequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT sequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 24h 2m 30s , 90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 12h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 24h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) );

SELECT sequal( spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 12h 2m 30s , 45d 0m 0s)' ) );

-- <> operator -----------------------

SELECT snotequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 24h 2m 30s , 90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 12h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 24h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) );

SELECT snotequal( spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 12h 2m 30s , 45d 0m 0s)' ) );

-- <-> operator ---------------------

SELECT sdist( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 0m 30s , 0d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) );

SELECT sdist( spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

