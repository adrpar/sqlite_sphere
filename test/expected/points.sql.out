1
0
1
0
1
0
1
(15.625d , 1.0416667d)
(0.625d , 0.041666667d)
(15.875d , -1.051d)
(0.875d , -0.051d)
(0.625d , 90d)
(0.625d , -90d)
( 15d 37m 30s , + 1d  2m 30s)
(  0d 37m 30s , + 0d  2m 30s)
( 15d 37m 30s , - 1d  2m 30s)
(  0d 37m 30s , - 0d  2m 30s)
(  0d 37m 30s , +90d  0m 0s)
(  0d 37m 30s , -90d  0m 0s)
(  1h  2m 30s , + 1d  2m 30s)
(  0h  2m 30s , + 0d  2m 30s)
(  1h  2m 30s , - 1d  2m 30s)
(  0h  2m 30s , - 0d  2m 30s)
(  0h  2m 30s , +90d  0m 0s)
(  0h  2m 30s , -90d  0m 0s)
(0 , 0)
(180.625d , 85d)
(0.625d , 5d)
(359.375d , -5d)
(180.625d , -85d)
(0.625d , -0.041666667d)
(0.625d , -0.041666667d)
(359.95833d , 0d)
(0 , 0)
(2.1415927 , 0)
0.00872664625996925
0.174532925199433
0.174532925199433
0.0
0.349065850398866
0.0
0.0
0.0109083078249646
3.15250096141476
0.0109083078249646
3.15250096141476
0.0109083078249646
0.0109083078249646
0.0109083078249646
0.0109083078249643
3.15250096141476
0.0109083078249643
3.15250096141476
0.0109083078249643
0.0109083078249643
0.0109083078249643
0.0
1.48352986419518
1.48352986419518
-1.48352986419518
-1.48352986419518
1.5707963267949
-1.5707963267949
0.0
1.48352986419518
1.48352986419518
-1.48352986419518
-1.48352986419518
1.5707963267949
-1.5707963267949
6.12323399573677e-17
0.0
-1.0
1
1
1
1
1
1
1
1
0
0
0
0
0
0
0
0
0
0
0
1
1
1
0.00872664625996925
0.174532925199433
0.174532925199433
0.0
0.349065850398866
0.0
0.0
